app/lib/efile/submission_error_parser.rb

```diff
@@ -6,23 +6,25 @@ def initialize(transition)
       @raw_response = transition.metadata["raw_response"]
     end
 
-    def to_xml
-      Nokogiri::XML(@raw_response)
-    end
-
-    def persist_efile_errors_from_transition_metadata
+    def persist_errors
```

Comment:
I did a little refactoring in this file. First, I made just this one method public and shortened its name -- I was getting really confused as to which method was being called in the state machine.

I also made the other methods that get called in persist_errors private and refactored the tests a bit.